File: src/main/kotlin/tonarinosibahaao/iobb/net/auth/servlet/SamlRequestServlet.java
package tonarinosibahaao.iobb.net.auth.servlet;

import com.onelogin.saml2.Auth;
import com.onelogin.saml2.exception.Error;
import com.onelogin.saml2.exception.SettingsException;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.Writer;

public class SamlRequestServlet extends HttpServlet {
    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        try {
            Auth auth = new Auth(req, resp);
            auth.login();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        Writer w = resp.getWriter();
        resp.setContentType("text/html");
        w.append("<html>" + "<head></head>"
                + "<body><h1>Click the button to send the AuthnRequest using HTTP POST</h1> <form method=\"POST\">"
                + "<input type=\"submit\" value=\"Send\"/>" + "</form>" + "</body>" + "</html>");
    }
}

File: src/main/kotlin/net/iobb/tonarinosibahaao/auth/servlet/SamlRequestServlet.kt
package net.iobb.tonarinosibahaao.auth.servlet

import com.onelogin.saml2.Auth
import java.io.Writer

import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

class SamlRequestServlet : HttpServlet(){

    override fun doGet(req: HttpServletRequest?, resp: HttpServletResponse) {
        try {
            val w: Writer = resp.writer
            resp.contentType = "text/html"
            w.append(
                "<html>" + "<head></head>"
                        + "<body><h1>Click the button to send the AuthnRequest using HTTP POST</h1> <form method=\"POST\">"
                        + "<input type=\"submit\" value=\"Send\"/>" + "</form>" + "</body>" + "</html>"
            )
        } catch (e: Exception) {
            e.printStackTrace()
        }

        val auth = Auth(req, resp)
        auth.login()

    }
}